### PR TITLE
refactor(rpc): migrate v07 rpc dtos to new serialization system

### PR DESCRIPTION
**Pull request by sistemd**
Progress on https://github.com/eqlabs/pathfinder/issues/1830.

Seemingly a lot of changes in this PR, but all of them are boilerplate. For now everything is tested with unit tests that we already had; what other ideas do you have? Do you even think more testing than that is necessary? Maybe I should run the load tests locally and see if any DTO failures come up?

### CODE REVIEW

**Comment by kkovaacs** [comment]
I had better luck with the starknet-js and starknet.py test suites, so I think we should run those with the new DTOs.

File: crates/rpc/src/method/call.rs [hunk 1]
```diff
@@ -0,0 +1,592 @@
...
+use crate::context::RpcContext;
+use crate::error::ApplicationError;
+
+#[derive(Debug)]
+pub enum CallError {
+    Internal(anyhow::Error),
+    Custom(anyhow::Error),
+    BlockNotFound,
+    ContractNotFound,
+    ContractErrorV05 { revert_error: String },
```
Comment (kkovaacs) [inline crates/rpc/src/method/call.rs:14]: BTW: this looks like a JSON-RPC 0.5 leftover. Do we still need this?

File: crates/rpc/src/method/simulate_transactions.rs [hunk 2]
```diff
@@ -210,117 +372,121 @@ pub(crate) mod tests {
...
-                            entry_point_type: EntryPointType::External,
-                            events: vec![OrderedEvent {
-                                order: 0,
-                                data: vec![
-                                    *account_contract_address.get(),
-                                    last_block_header.sequencer_address.0,
-                                    Felt::from_u64(OVERALL_FEE),
-                                    felt!("0x0"),
+            result.serialize(Serializer {
+                version: RpcVersion::V06,
```
Comment (kkovaacs) [inline crates/rpc/src/method/simulate_transactions.rs:376]: Why are we using `V06` here?

File: crates/rpc/src/method/simulate_transactions.rs [hunk 3]
```diff
@@ -1436,8 +1602,12 @@ pub(crate) mod tests {
         let result = simulate_transactions(context, input).await.unwrap();
 
         pretty_assertions_sorted::assert_eq!(
-            result,
-            SimulateTransactionOutput(vec![
+            result
+                .serialize(Serializer {
+                    version: RpcVersion::V06
```
Comment (kkovaacs) [inline crates/rpc/src/method/simulate_transactions.rs:1607]: Same question: I think these serializers reflect the 0.7 JSON-RPC spec, so we should probably be using `V07` here? (I understand that it doesn't really make a difference right now.)

**Review by kkovaacs — APPROVED** [review]
LGTM except for a few trivial nits.

**Review by CHr15F0x — APPROVED** [review]

**Pull request merged**